klog.go

```diff
@@ -917,7 +917,14 @@ func LogToStderr(stderr bool) {
 
 // output writes the data to the log files and releases the buffer.
 func (l *loggingT) output(s severity, log *logr.Logger, buf *buffer, depth int, file string, line int, alsoToStderr bool) {
+	var isLocked = true
 	l.mu.Lock()
+	defer func() {
+		if isLocked {
+			l.mu.Unlock() // Make sure to unlock if panic occurs below
```

Comment:
A panic is not the only reason why we get here. A normal function exit also comes through here.

```suggestion
			// Unlock before returning in case that it wasn't done already.
			l.mu.Unlock()
```

Reply:
Updated, PTAL thanks.